dbt_semantic_interfaces/validations/primary_entity.py

```diff
@@ -0,0 +1,101 @@
+import logging
+from typing import Generic, List, Sequence
+
+from dbt_semantic_interfaces.implementations.semantic_manifest import (
+    PydanticSemanticManifest,
+)
+from dbt_semantic_interfaces.protocols import Entity, SemanticManifestT, SemanticModel
+from dbt_semantic_interfaces.references import SemanticModelReference
+from dbt_semantic_interfaces.type_enums import EntityType
+from dbt_semantic_interfaces.validations.validator_helpers import (
+    FileContext,
+    SemanticManifestValidationRule,
+    SemanticModelContext,
+    ValidationError,
+    ValidationIssue,
+    validate_safely,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class PrimaryEntityRule(SemanticManifestValidationRule[SemanticManifestT], Generic[SemanticManifestT]):
+    """If a semantic model contains dimensions, the primary entity must be available.
+
+    The primary entity could be defined by the primary_entity field, or by one of the entities defined in a semantic
+    model.
+    """
+
+    @staticmethod
+    def _model_requires_primary_entity(semantic_model: SemanticModel) -> bool:
+        return len(semantic_model.dimensions) > 0
+
+    @staticmethod
+    def _model_has_entity_with_primary_type(semantic_model: SemanticModel) -> bool:
+        return any(entity.type == EntityType.PRIMARY for entity in semantic_model.entities)
+
+    @staticmethod
+    def _entity_with_primary_type_in_model(semantic_model: SemanticModel) -> Entity:
+        for entity in semantic_model.entities:
+            if entity.type == EntityType.PRIMARY:
+                return entity
+        entities_with_primary_type = tuple(
+            entity for entity in semantic_model.entities if entity.type == EntityType.PRIMARY
+        )
+
+        if len(entities_with_primary_type) == 1:
+            return entities_with_primary_type[0]
+
+        raise RuntimeError(f"Did not find exactly one entity with entity type {EntityType.PRIMARY} in {semantic_model}")
+
+    @staticmethod
+    def _check_model(semantic_model: SemanticModel) -> Sequence[ValidationIssue]:
```

Comment:
This method should also have the `@validate_safely` decorator. By having it just on the base `validate_manifest` if one semantic model has an issue, any semantic models to be checked after it won't get checked because the exception handling happens outside of the for loop iterating over the models. By adding `@validate_safely` on `_check_model` it ensures that if an exception is raised for a semantic_model, it only stops this validation step for that semantic_model.

Reply:
Updated.